mungegithub/mungers/approval-handler.go

```diff
@@ -90,8 +96,88 @@ func (h *ApprovalHandler) Munge(obj *github.MungeObject) {
 	}
 
 	approverSet := sets.String{}
+	if obj.Issue.User.Name != nil {
+		approverSet.Insert(*obj.Issue.User.Name)
+	}
+
+	approverSet = approverSet.Union(createApproverSet(comments))
+	needsApproval := h.getApprovalNeededFiles(files, approverSet)
+
+	if needsApproval.Len() > 0 {
+		glog.Infof("Canceling the approve label because these files %v have no valid approvers", needsApproval)
+		//if latest notification stale, update it
+		if notificationNeedsUpdate(obj) {
```

Comment:
By reading this code, I don't know if it removes it or not. Maybe rename the function?